wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala

```diff
@@ -0,0 +1,279 @@
+package org.bitcoins.wallet
+import org.bitcoins.core.crypto._
+import org.bitcoins.core.crypto.bip44._
+import org.bitcoins.core.currency.{CurrencyUnit, Satoshis}
+import org.bitcoins.core.number.UInt32
+import org.bitcoins.core.protocol.BitcoinAddress
+import org.bitcoins.core.protocol.blockchain.ChainParams
+import org.bitcoins.core.protocol.transaction.{
+  Transaction,
+  TransactionOutPoint,
+  TransactionOutput
+}
+import org.bitcoins.core.util.{BitcoinSLogger, EitherUtil}
+import org.bitcoins.core.wallet.builder.BitcoinTxBuilder
+import org.bitcoins.core.wallet.fee.SatoshisPerByte
+import org.bitcoins.core.wallet.utxo.BitcoinUTXOSpendingInfo
+import org.bitcoins.db.DbConfig
+import org.bitcoins.wallet.api._
+import org.bitcoins.wallet.models._
+import scodec.bits.BitVector
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+
+sealed abstract class Wallet extends UnlockedWalletApi with BitcoinSLogger {
+  implicit def ec: ExecutionContext
+
+  val addressDAO: AddressDAO = AddressDAO(dbConfig)
+  val mnemonicDAO: MnemonicCodeDAO = MnemonicCodeDAO(dbConfig)
+  val accountDAO: AccountDAO = AccountDAO(dbConfig)
+  val utxoDAO: UTXOSpendingInfoDAO = UTXOSpendingInfoDAO(dbConfig)
+
+  val bip44Coin: BIP44Coin = BIP44Coin.fromChainParams(chainParams)
+
+  def dbConfig: DbConfig
+
+  def chainParams: ChainParams
+
+  val feeProvider: FeeProvider = _ =>
+    Future.successful(SatoshisPerByte(Satoshis.one))
+
+  /**
+    * @inheritdoc
+    */
+  override def lock: Future[WalletApi] = ???
+
+  /**
+    * @inheritdoc
+    */
+  override def addUtxo(
```

Comment:
I think you've done a nice job of sectioning off this method into logical pieces, I think it would be nice to make them separate defs to make this easier to reason about.